### PR TITLE
Various improvements

### LINKED ISSUE #5: double icons with github refined plugin
double icons with the github refined plugin ( https://github.com/sindresorhus/refined-github )

### PR DESCRIPTION
**Pull request by kidonng**
- Drop jQuery
- Fix compatibility with Refined GitHub's [`edit-files-faster`](https://github.com/sindresorhus/refined-github/blob/master/source/features/edit-files-faster.tsx) feature, Fix #5
- Use `selector-observer` to replace GitHub file tree icons (we are using this package in Refined GitHub and it's better than manually create a MutationObserver)

### CODE REVIEW

File: src/content.js [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-  observer.observe(document.body, {
-    attributes: true,
-    childList: true,
-    characterData: true,
-    subtree: true
-  });
 
-  injectGithub('body');
-}
+      icon.classList.add('octicon-file', 'gfi', className);
```
Comment (lvarayut) [inline src/content.js:19]: When the `isColor` is set to true, the `className` could contain multiple classes separated by space, i.e. "github-icon medium-red" which causes the following error:

```
 Failed to execute 'add' on 'DOMTokenList': The token provided ('git-icon medium-red') contains HTML space characters, which are not valid in tokens
```

Therefore, it needs to be split by space and spread to the `classist.add` function

File: src/content.js [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+      item.querySelector('svg').replaceWith(icon);
+    }
+  }
 
-document.addEventListener('pjax:end', init);
-init();
+  observe('.js-navigation-container > .js-navigation-item', {
+    add(element) {
+      injectGithub(element);
+    }
```
Comment (fregante) [inline src/content.js:27]: This could be:
```suggestion
    add: injectGithub
```

File: src/content.js (hunk 1 above)
Comment (kidonng) [reply src/content.js:19]: I didn't know how to switch color so I didn't test it. Thanks for the reminder!

File: src/content.css [hunk 3]
```diff
@@ -1,8 +1,9 @@
 .gfi {
   color: rgba(3,47,98,0.55);
+  display: inline-block;
+  vertical-align: top;
 }
 
 td.icon > svg.octicon-file, .rgh-edit-files-faster {
   display: none;
 }
```
Comment (lvarayut) [inline src/content.css:9]: According to your [new implementation of the icons](https://github.com/lvarayut/github-file-icons/pull/17/files#diff-95ffcbff3c841ad0a802793cabd8940eR9-R19), the icons will be shown inside the `.rgh-edit-files-faster`. Hence, the following code needs to be deleted in order to display the icons correctly:

```
td.icon > svg.octicon-file, .rgh-edit-files-faster {
 display: none;
}
```

**Review by lvarayut — CHANGES_REQUESTED** [review]
There is an issue regarding the `display` property in CSS, which should be removed entirely. After it has been removed, the PR should be good to go.

File: src/content.css (hunk 3 above)
Comment (kidonng) [reply src/content.css]: Ah, I was wondering why the icons didn't show up yesterday (I initially made this PR with Refined GitHub disabled so I didn't notice the issue). Nice catch!

**Comment by lvarayut** [comment]
Thanks a lot @kidonng for the awesome PR! 🚀

**Pull request merged**